### PR TITLE
Prevent running with local ZenStore and remote code execution

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
This PR adds an error if trying to run with a remote orchestrator or step operator in combination with a local ZenStore.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/stack/stack.py [hunk 1]
```diff
@@ -660,6 +678,15 @@ def prepare_pipeline_deployment(
                     f"    `zenml stack up`\n"
                 )
 
+        if self.requires_remote_server and Client().zen_store.is_local_store():
+            raise RuntimeError(
+                "Remote orchestrators and step operators require a remote "
+                "ZenML server. To run a pipeline with this stack you need to "
+                "connect to a remote ZenML server first. Check out "
+                "https://docs.zenml.io/getting-started/deploying-zenml for "
+                "more information on how to deploy ZenML."
```
Comment (htahir1) [inline src/zenml/stack/stack.py:687]: Please feel free to reformat in your IDE. My suggestion is more on wording.

```suggestion
                "Stacks with remote components such as remote orchestrators "
                "and step operators require a remote "
                "ZenML server. To run a pipeline with this stack you need to "
                "connect to a remote ZenML server first. Check out "
                "https://docs.zenml.io/getting-started/deploying-zenml for "
                "more information on how to deploy ZenML."
```

File: src/zenml/stack/stack.py [hunk 2]
```diff
@@ -530,6 +531,21 @@ def setting_classes(self) -> Dict[str, Type["BaseSettings"]]:
...
+    def requires_remote_server(self) -> bool:
+        """If the stack requires a remote ZenServer to run.
+
+        This is the case if any code is getting executed remotely. This is the
+        case for both remote orchestrators as well as remote step operators.
+
+        Returns:
+            If the stack requires a remote ZenServer to run.
+        """
+        return self.orchestrator.config.is_remote or (
```
Comment (htahir1) [inline src/zenml/stack/stack.py:544]: In what other cases do you think in the future we can throw this error?

**Review by htahir1 — CHANGES_REQUESTED** [review]
Small PR

**Review by AlexejPenner — APPROVED** [review]
🦭 of approval - pending Passing CI

**Review by safoinme — APPROVED** [review]
LGTM

**Review by htahir1 — APPROVED** [review]

**Pull request merged**